### PR TITLE
Pull Request for Issue1915: Carbon filter farm refactor

### PR DESCRIPTION
**Pull request by helfrij**
- Removed BioXASCarbonFilterFarmActuatorControl, moved basic functionality to subclass BioXASCarbonFilterFarmActuator. Actuator now inherits from BioXASBeamlineComponent.
- Removed BioXASCarbonFilterFarmControl, moved basic functionality to subclass BioXASCarbonFilterFarm. FilterFarm now inherits from BioXASBeamlineComponent.
- Renamed BioXASCarbonFilterFarmActuatorControlView to BioXASCarbonFilterFarmActuatorView.
- Renamed BioXASCarbonFilterFarmControlView to BioXASCarbonFilterFarmView.

### CODE REVIEW

**Comment by helfrij** [comment]
Hey, @iainworkman. Would you review this pull request?

**Comment by iainworkman** [comment]
Looks good to me.
Time Estimate: 15 minutes

File: source/beamline/BioXAS/BioXASCarbonFilterFarmActuator.cpp [hunk 1]
```diff
@@ -13,43 +40,117 @@ BioXASCarbonFilterFarmActuator::~BioXASCarbonFilterFarmActuator()
...
 	return result;
 }
 
-void BioXASCarbonFilterFarmActuator::addWindow(Window::Option window, double positionSetpoint, double positionMin, double positionMax, double filter)
+bool BioXASCarbonFilterFarmActuator::setPositionStatus(AMControl *newControl)
+{
+	bool result = false;
+
+	if (position_) {
+		position_->setStatus(newControl);
```
Comment (dretrex) [inline source/beamline/BioXAS/BioXASCarbonFilterFarmActuator.cpp]: Does `setStatus()` return a bool?  It would make sense within your paradigm.

**Comment by dretrex** [comment]
Looks good to me too.  The comment I made was more of a general comment than a request for change.
Time Estimate: 10 minutes

**Comment by dretrex** [comment]
And don't forget to set the Estimate when making the pull request ready for review.

**Pull request merged**